src/core/tests/type_prop/arithmetic_ops.hpp

```diff
@@ -114,9 +117,68 @@ TYPED_TEST_P(ArithmeticOperator, shape_inference_4D_x_3D_numpy_broadcast) {
 
     const auto op = std::make_shared<TypeParam>(A, B);
 
-    ASSERT_EQ(op->get_element_type(), element::f32);
-    ASSERT_EQ(op->get_shape(), (Shape{8, 7, 6, 5}));
-    ASSERT_EQ(op->get_autob(), op::AutoBroadcastType::NUMPY);
+    EXPECT_EQ(op->get_element_type(), element::f32);
+    EXPECT_EQ(op->get_shape(), (Shape{8, 7, 6, 5}));
+    EXPECT_EQ(op->get_autob(), op::AutoBroadcastType::NUMPY);
+}
+
+TYPED_TEST_P(ArithmeticOperator, shape_inference_pdpd_doc_examples) {
+    {
+        auto A = std::make_shared<op::Parameter>(element::f32, Shape{2, 3, 4, 5});
+        auto B = std::make_shared<op::Parameter>(element::f32, Shape{3, 4});
+
+        const auto autob = op::AutoBroadcastSpec(op::AutoBroadcastType::PDPD, 1);
+        const auto op = std::make_shared<TypeParam>(A, B, autob);
+
+        EXPECT_EQ(op->get_element_type(), element::f32);
+        EXPECT_EQ(op->get_shape(), (Shape{2, 3, 4, 5}));
+        EXPECT_EQ(op->get_autob().m_type, op::AutoBroadcastType::PDPD);
+    }
+    {
```

Comment:
Make separate test case for each block. 
Review similar cases in the tests

Reply:
Okay, done